api/pkg/service/auth/service.go

```diff
@@ -66,16 +75,22 @@ func (v *Validator) JWTAuth(ctx context.Context, jwt string, scheme *security.JW
 	return WithUserID(ctx, uint(userID)), nil
 }
 
-// UserFromContext fetch user id from the passed context verfies if it exists in db
+// User fetch user id from the passed context verfies if it exists in db
 // returns the User object
-func (v *Validator) UserFromContext(ctx context.Context) (*model.User, error) {
+func (v *Service) User(ctx context.Context) (*model.User, error) {
+
+	logger := v.Logger(ctx)
+	db := v.DB(ctx)
+
 	userID := UserID(ctx)
-	log := v.Logger.With("user-id", userID)
+	log := &log.Logger{
+		SugaredLogger: logger.With(zap.String("user-id", strconv.FormatUint(uint64(userID), 10))),
+	}
```

Comment:
I think we would need to override the method `With` in `log.Logger` so that  services can use `With` to add additional fields.